arch/xtensa/core/cpu_idle.c

```diff
@@ -6,48 +6,14 @@
 #include <zephyr/toolchain.h>
 #include <zephyr/tracing/tracing.h>
 
-/* xt-clang removes any NOPs more than 8. So we need to set
- * no optimization to avoid those NOPs from being removed.
- *
- * This function is simply enough and full of hand written
- * assembly that optimization is not really meaningful
- * anyway. So we can skip optimization unconditionally.
- * Re-evalulate its use and add #ifdef if this assumption
- * is no longer valid.
- */
+#ifndef CONFIG_ARCH_CPU_IDLE_CUSTOM
 __no_optimization
 void arch_cpu_idle(void)
 {
 	sys_trace_idle();
-
-	/* Just spin forever with interrupts unmasked, for platforms
-	 * where WAITI can't be used or where its behavior is
-	 * complicated (Intel DSPs will power gate on idle entry under
-	 * some circumstances)
-	 */
-	if (IS_ENABLED(CONFIG_XTENSA_CPU_IDLE_SPIN)) {
-		__asm__ volatile("rsil a0, 0");
-		__asm__ volatile("loop_forever: j loop_forever");
-		return;
-	}
-
-	/* Cribbed from SOF: workaround for a bug in some versions of
-	 * the LX6 IP.  Preprocessor ugliness avoids the need to
-	 * figure out how to get the compiler to unroll a loop.
-	 */
-	if (IS_ENABLED(CONFIG_XTENSA_WAITI_BUG)) {
-#define NOP4 __asm__ volatile("nop; nop; nop; nop");
-#define NOP32 NOP4 NOP4 NOP4 NOP4 NOP4 NOP4 NOP4 NOP4
-#define NOP128() NOP32 NOP32 NOP32 NOP32
-		NOP128();
-#undef NOP128
-#undef NOP32
-#undef NOP4
-		__asm__ volatile("isync; extw");
-	}
-
 	__asm__ volatile ("waiti 0");
 }
+#endif
```

Comment:
This also breaks git bisect as if you take this first commit of the PR and build for cAVS2.5, you get a broken binary as WAITI_BUG workaround is not correctly applied.

Reply:
Ok, I will change the order. I tested it and it didn't break cavs2.5